### PR TITLE
docs: clarify where `--paths` option updates in spec file

### PR DESCRIPTION
**Pull request by hkennyv**
Hi there, I would love a review to get some thoughts on this PR and maybe some word-smithing. This addresses #5961.

### CODE REVIEW

**Comment by bwoodsend** [comment]
Well this is your call. If that's really all you would have needed to avoid your confusion over on #5961 then great - I approve.

I'd have thought that you'd want something going the other way i.e. somewhere in the [spec files page](https://pyinstaller.readthedocs.io/en/latest/spec-files.html) say that `pathex` corresponds to the `--paths` CLI option but that kind of information would be best suited to go in an API reference for the Analysis class which we currently lack...

**Comment by bwoodsend** [comment]
Would it help if it went in [the CLI reference](https://pyinstaller.readthedocs.io/en/latest/usage.html#cmdoption-p) too and `pyinstaller --help` (currently defined [here](https://github.com/pyinstaller/pyinstaller/blob/0f2b2e921433ab5a510c7efdb21d9c1d7cfbc645/PyInstaller/building/makespec.py#L260))?

**Comment by hkennyv** [comment]
@bwoodsend hey thanks for the response. I agree with you and think the best solution is to include it in all 3 places: the CLI reference, using spec files, and when things go wrong. I'm happy to contribute those myself as part of this PR as well.

Maybe it is just the way I do things, but my approach to documentation is normally to go through the usage guides or examples first and I won't go through the API reference until I need to. So I think it is still worth clarifying on the "When things go wrong" and "Using spec files" pages since they're more likely to be read compared to the API reference, assuming others operate this same way :p

**Comment by bwoodsend** [comment]
Go for it then. Put it anywhere you looked trying to find it.

**Comment by hkennyv** [comment]
@bwoodsend thanks! i'll try to get to it in the next couple days

**Comment by hkennyv** [comment]
@bwoodsend apologies for the delayed update. that should cover all 3: CLI reference, Using spec files page, and the When things go wrong page

File: doc/spec-files.rst [hunk 1]
```diff
@@ -97,6 +97,8 @@ The statements in a spec file create instances of four classes,
 
   - ``scripts``: the python scripts named on the command line;
   - ``pure``: pure python modules needed by the scripts;
+  - ``pathex``: a list of paths to search for imports (like using PYTHONPATH),
+    including paths given by the :option:`--paths` option.
```
Comment (bwoodsend) [inline doc/spec-files.rst:101]:
```suggestion
  - ``pathex``: a list of paths to search for imports (like using
    :envvar:`PYTHONPATH`), including paths given by the :option:`--paths`
    option.
```

**Review by bwoodsend — APPROVED** [review]
Looks good to me!

**Comment by bwoodsend** [comment]
> BTW, are there any plans to add an API reference for the Analysis, EXE classes, etc? I think that would be a great supplement to the Using spec files page.

See you're kinda asking the wrong question. Are there plans? Yes there are. But are they likely to happen any time soon? No.

**Comment by hkennyv** [comment]
> > BTW, are there any plans to add an API reference for the Analysis, EXE classes, etc? I think that would be a great supplement to the Using spec files page.
> 
> See you're kinda asking the wrong question. Are there plans? Yes there are. But are they likely to happen any time soon? No.

Is there anything we (contributors) can help with to push this forward?

**Comment by bwoodsend** [comment]
Well you can have a go at writing if you're determined but we haven't really thought it through in much detail.

**Comment by hkennyv** [comment]
@bwoodsend I can surely give it a shot. Although I think that that would warrant its own PR.

**Comment by bwoodsend** [comment]
> Although I think that that would warrant its own PR.

Definitely.

**Pull request merged**